test/actions/recover.js

```diff
@@ -0,0 +1,225 @@
+import test from 'ava';
+import nock from 'nock';
+
+import configureMockStore from 'redux-mock-store';
+import thunk from 'redux-thunk';
+
+import {
+  applyRecoveryWorkflowToCollection,
+  applyRecoveryWorkflowToGranule
+} from '../../app/scripts/actions';
+import {
+  constructCollectionNameVersion
+} from '../../app/scripts/utils/format';
+import { requestMiddleware } from '../../app/scripts/middleware/request';
+
+const middlewares = [
+  requestMiddleware,
+  thunk
+];
+const mockStore = configureMockStore(middlewares);
+const store = mockStore({
+  granules: {
+    list: {
+      data: [],
+      meta: {},
+      params: {}
+    },
+    recovered: {},
+    executed: {},
+    meta: {}
+  }
+});
+const getActionType = (action) => action.type;
+const name = 'fakeCollection';
+const version = '000';
+const collectionId = constructCollectionNameVersion(name, version);
+const granuleId = 'fakeId';
+
+test.beforeEach((t) => {
+  t.context.defaultConfig = {
+    json: true,
+    resolveWithFullResponse: true,
+    simple: false
+  };
+  store.clearActions();
+});
+
+test.serial('appylyRecoveryWorkflowToCollection fails to acquire collection, dispatches COLLECTION_APPLYWORKFLOW_ERROR', async (t) => {
+  nock('https://example.com')
```

Comment:
these `example.com`s should probably be `localhost:3000`

Reply:
We're using `nock` to make sure these tests don't go to network, right? So yeah, it seems like they should match whatever URL the components will actually try to request?

Reply:
My unit tests by default (`npx ava test/actions/recover.js`) are using the `example.com` endpoint. That doesn't seem to be the case for other tests, so I need to figure out what's going on there.

Reply:
`test/actions/store.js` nocks `https://localhost:5001`, but it's also defining it's `requestAction` such that the `url` matches:
```
const requestAction = {
  type: 'TEST',
  method: 'GET',
  url: 'http://localhost:5001/test-path'
```

The default `apiRoot` configuration for the dashboard points to `example.com` which is why my tests are pointing to `example.com`.